org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/JavadocCompletionProposal.java

```diff
@@ -117,6 +118,10 @@ public List<CompletionItem> getProposals(ICompilationUnit cu, int offset, Comple
 			ci.setSortText(SortTextHelper.convertRelevance(0));
 			ci.setKind(CompletionItemKind.Snippet);
 			ci.setInsertTextFormat(isSnippetSupported ? InsertTextFormat.Snippet : InsertTextFormat.PlainText);
+			if (JavaLanguageServerPlugin.getPreferencesManager().getClientPreferences().isCompletionItemInsertTextModeSupport() &&
```

Comment:
For javadoc completion items, shouldn't we also check whether `itemDefaults` already has `insertTextMode` set ?

Reply:
`ItemDefaults` is not used for JavaDoc completion since there is only one result, just checking that `InsertTextMode` is supported here.

Reply:
One question:

It's true that `JavadocCompletionProposal` will only return upmost one completion item. But according to 
https://github.com/eclipse/eclipse.jdt.ls/blob/2b3235286b0c77b3a0766c5e8ae83ef6dd309f46/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandler.java#L266

The result will be merged to the main list in `CompletionHandler`. Seems that according to the code there, we are not guaranteed that the item returned by `JavadocCompletionProposal` is the only one in the list?

Reply:
Right, there are cases where there are other `completionItems` are in the `completionList` along with javadoc comment, my thought process was that since `JavadocCompletionProposal` always only returns one result, there wasn't much point to implement itemDefaults for it. But for the sake of consistency I can add that to `JavadocCompletionProposal` as well?

Reply:
I prefer to add that for consistency. Then I think you can call the utility method extracted before and the code will become cleaner.